--- a/rns_shell/compositor/Compositor.h	2022-04-01 14:29:42.000000000 +0530
+++ b/rns_shell/compositor/Compositor.h	2022-04-17 11:02:33.275906873 +0530
@@ -7,6 +7,8 @@
 */
 #pragma once

+#include "ReactSkia/utils/RnsUtils.h"
+
 #include "third_party/skia/include/core/SkRect.h"

 #include "WindowContext.h"
@@ -62,7 +64,7 @@
     SharedLayer rootLayer_;
     std::unique_ptr<WindowContext> windowContext_;
     sk_sp<SkSurface> backBuffer_;
-    uint64_t nativeWindowHandle_;
+    GLNativeWindowType nativeWindowHandle_;

 #if USE(RNS_SHELL_PARTIAL_UPDATES)
     bool supportPartialUpdate_;
--- a/rns_shell/common/WindowContext.h	2022-04-01 14:29:42.000000000 +0530
+++ b/rns_shell/common/WindowContext.h	2022-04-17 11:01:57.835622761 +0530
@@ -24,6 +24,13 @@

 namespace RnsShell {

+#if USE(EGL)
+#include <EGL/egl.h>
+typedef EGLNativeWindowType GLNativeWindowType;
+#else
+typedef uint64_t GLNativeWindowType;
+#endif
+
 class WindowContext {
 public:
     WindowContext(const DisplayParams&);
--- a/rns_shell/platform/graphics/gl/GLWindowContext.h	2022-04-01 14:29:42.000000000 +0530
+++ b/rns_shell/platform/graphics/gl/GLWindowContext.h	2022-04-17 10:59:40.314672289 +0530
@@ -37,12 +37,6 @@
 #include "PlatformDisplay.h"
 #include "WindowContext.h"

-#if USE(EGL)
-typedef EGLNativeWindowType GLNativeWindowType;
-#else
-typedef uint64_t GLNativeWindowType;
-#endif
-
 namespace RnsShell {

 class GLWindowContext : public WindowContext {
--- a/rns_shell/platform/graphics/WindowContextFactory.h	2022-04-01 14:29:42.000000000 +0530
+++ b/rns_shell/platform/graphics/WindowContextFactory.h	2022-04-17 10:59:17.950547150 +0530
@@ -18,12 +18,8 @@

 #if USE(EGL)
 #include <EGL/eglplatform.h>
-typedef EGLNativeWindowType GLNativeWindowType;
 #elif USE(GLX)
 #include <GL/glx.h>
-typedef uint64_t GLNativeWindowType;
-#else // !USE(EGL)
-typedef uint64_t GLNativeWindowType;
 #endif

 // webgpu_cpp.h and X.h don't get along. Include this first, before X11 defines None, Success etc.
@@ -33,6 +29,7 @@

 #include "ReactSkia/utils/RnsLog.h"
 #include "PlatformDisplay.h"
+#include "WindowContext.h"
 #if PLATFORM(X11)
 #include "x11/PlatformDisplayX11.h"
 #endif
